File: src/ReSharperExtension/AboutAction.cs
using System.Windows.Forms;
using JetBrains.ActionManagement;
using JetBrains.Application.DataContext;

namespace ReSharperExtension
{
    [ActionHandler("ReSharperExtension.About")]
    public class AboutAction : IActionHandler
    {
        public bool Update(IDataContext context, ActionPresentation presentation, DelegateUpdate nextUpdate)
        {
            // return true or false to enable/disable this action
            return true;
        }

        public void Execute(IDataContext context, DelegateExecute nextExecute)
        {
            MessageBox.Show(
              "2\nAcme Corp.\n\n",
              "About 2",
              MessageBoxButtons.OK,
              MessageBoxIcon.Information);
        }
    }
}

File: src/Calc/AboutAction.fs
module AboutAction

//open JetBrains.ActionManagement
//open JetBrains.Application
//open JetBrains.UI.ActionsRevised
//open JetBrains.UI.Application
//open JetBrains.UI.ToolWindowManagement
//open System.Windows.Forms

//[<assembly: 
//("YC.ReSharper.AbstractAnalysis.Languages.Calc.Actions.xml")>]
//do ()
//
////[<("Plugins.Calc.About")>]
//[<ToolWindowDescriptor(Text = "Calc", 
//    VisibilityPersistenceScope = ToolWindowVisibilityPersistenceScope.Solution,
//    Type = ToolWindowType.SingleInstance,
//    InitialDocking = ToolWindowInitialDocking.NotSpecified)>]
//type WindowDescriptor (applicationHost : IApplicationHost) =
//    inherit ToolWindowDescriptor(applicationHost)
//
//type AboutAction () =
//    inherit ActivateToolWindowActionHandler<WindowDescriptor>()
//        member this.Update (context, presentation, nextUpdate) = true
//        member this.Execute (context, nextExecute) =
//            MessageBox.Show("Calc plugin") |> ignore